cli/src/pcluster/api/controllers/common.py

```diff
@@ -51,7 +51,19 @@ def configure_aws_region(is_query_string_arg: bool = True):
     def _decorator_validate_region(func):
         @functools.wraps(func)
         def _wrapper_validate_region(*args, **kwargs):
-            region = kwargs.get("region") if is_query_string_arg else request.get_json().get("region")
+            region = None
+            # the region may either be specified in the query parameters, in
+            # which case we can simply look into the keyword arguments.
+            # Otherwise, it may be specified in the body -- which may either be
+            # in the flask request if called via flask or in the body which is
+            # the fisrt arg
+            if is_query_string_arg:
+                region = kwargs.get("region")
+            elif request:
+                region = request.get_json().get("region")
+            elif len(args) > 0:
+                region = args[0].get("region")
```

Comment:
Since this only needs to be done in a few places that seems like a good compromise. I updated the logic so that if you pass a `body_name` it will look that up in the kwargs (or in the request json) -- otherwise it checks the query parameters. The CLI tests all seem to pass with this while maintaining generality from the generic operations (along with adding the body to the kwargs where we convert the arguments in `entrypoint`).

Reply:
This change does mean that we can’t get it from the first arg now if we were to try to call directly— my guess is that shouldn’t happen (though by that argument the old mechanism was safe too). Originally I had code to check  both but this seems fine.

Note that it's also possible to get the arg name through inspection -- it makes the interface cleaner but implementation more complicated. I'm fine with how it is